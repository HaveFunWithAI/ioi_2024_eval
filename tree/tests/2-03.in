2000
0 1 2 3 4 5 6 7 8 9 10 10 10 13 14 15 16 15 13 16 17 18 22 22 23 25 25 25 20 29 29 31 32 33 31 35 35 37 38 38 37 41 38 43 43 42 42 47 37 49 45 51 49 52 54 55 55 56 58 59 56 61 54 52 36 58 54 67 67 69 69 71 69 73 73 72 76 74 78 79 80 81 82 83 84 80 84 87 88 89 84 91 78 93 93 93 96 96 98 99 99 101 94 82 89 105 98 107 108 97 105 93 112 112 114 115 93 101 118 119 120 121 122 123 123 122 126 125 124 127 130 124 125 123 134 135 134 137 134 139 140 138 142 142 143 145 145 147 140 149 150 150 152 152 154 151 156 153 136 159 160 161 161 162 164 163 160 167 168 168 170 168 130 165 137 175 176 175 123 179 180 181 181 183 182 181 158 187 188 189 190 171 192 193 194 195 194 197 162 199 195 200 162 203 203 205 204 207 208 209 203 211 211 204 210 212 216 217 209 214 119 221 222 174 82 99 226 227 228 227 227 231 79 88 159 184 236 236 236 239 239 240 242 243 244 243 246 236 248 248 249 219 252 252 134 251 256 237 258 259 259 261 261 263 263 265 260 262 268 268 269 259 272 273 272 274 268 277 274 274 280 281 282 270 284 285 286 274 268 263 79 291 292 293 294 295 294 297 295 297 297 301 300 303 304 304 301 307 307 309 310 311 312 308 314 315 316 316 318 319 313 319 320 323 324 324 326 325 324 329 330 329 325 333 324 335 336 324 330 339 339 330 324 343 333 344 346 346 348 348 348 349 352 352 354 354 355 354 358 359 348 361 362 347 364 333 366 366 367 369 370 369 371 373 374 375 376 374 378 375 375 381 381 383 378 376 377 378 388 369 371 391 391 393 391 393 391 397 398 399 379 324 363 334 359 328 406 407 408 409 409 410 412 413 413 413 350 318 307 419 420 420 422 423 423 283 426 427 428 429 430 429 429 430 431 435 436 436 438 430 440 428 442 443 430 445 446 445 448 449 449 450 452 451 454 433 339 457 458 269 460 461 462 461 463 461 199 467 463 469 446 137 472 473 474 473 476 476 478 478 480 478 482 477 483 485 485 487 485 489 486 491 479 493 316 251 496 497 340 499 499 500 502 502 504 505 502 507 508 509 509 505 512 508 514 515 516 515 514 519 520 519 522 523 521 525 525 527 518 509 530 531 505 48 534 534 536 534 538 534 350 541 541 171 255 545 545 547 545 549 550 551 550 552 547 548 556 556 558 559 558 557 562 560 562 555 566 567 568 569 570 570 548 565 552 575 576 577 578 577 580 581 581 581 584 580 586 587 588 589 590 587 590 593 585 590 596 597 598 599 600 601 601 598 602 600 563 607 608 609 600 563 573 613 323 615 616 617 618 618 117 621 622 621 623 625 626 627 628 626 628 631 632 627 630 632 636 637 638 639 637 318 642 643 643 645 645 644 645 649 644 651 651 652 651 644 650 657 658 642 658 661 662 663 664 665 662 644 644 669 670 671 672 672 669 674 670 674 377 121 631 681 409 129 684 685 329 572 350 374 690 124 666 693 694 695 696 569 698 623 700 700 702 169 316 705 705 707 708 707 710 709 709 713 461 218 566 453 718 207 114 434 722 722 724 389 726 52 728 36 730 731 730 731 734 302 279 449 738 738 740 739 740 740 744 739 739 747 747 739 747 751 752 747 754 754 701 660 478 759 760 628 762 763 764 764 766 767 768 769 612 771 772 772 657 775 776 775 776 779 780 780 775 783 775 777 777 787 788 786 790 791 792 793 794 795 792 797 790 798 797 801 794 795 804 804 570 807 807 809 809 811 809 813 714 151 399 817 818 817 819 322 822 823 824 822 823 827 824 829 295 831 832 833 833 835 836 834 833 335 840 561 589 843 844 241 846 182 848 849 848 720 569 853 290 855 855 857 858 855 860 751 862 863 392 865 865 867 867 869 867 866 872 872 874 869 869 877 878 876 880 873 869 883 883 883 886 887 871 889 890 213 322 893 274 895 895 897 895 899 900 895 902 902 902 904 906 901 898 909 909 911 912 911 910 915 916 915 916 917 916 921 921 917 919 925 926 920 927 929 930 931 929 931 917 934 932 937 930 939 939 911 933 933 944 945 945 947 944 949 949 951 952 952 952 955 956 956 956 957 960 961 960 963 963 965 963 967 963 960 970 964 968 959 961 975 976 975 978 976 980 955 959 983 225 418 986 438 410 304 380 991 992 993 993 993 992 641 43 840 1000 1000 1002 1002 1003 1002 1006 1007 1008 1008 1001 1011 1012 1013 1014 1012 1015 1012 1018 1019 1019 1021 1022 1023 1024 1006 1026 1026 1028 1029 1030 1031 1031 1031 1028 1035 1036 1037 1038 1035 1040 1041 1042 1043 1001 1045 1045 1047 1046 1047 1050 1051 1052 1052 1052 1054 1056 1057 1013 1011 374 722 831 1063 1064 1065 1066 1066 1064 1064 1070 1070 1072 1073 1074 1074 1075 1077 1078 1070 1080 1081 1081 1083 1083 1085 1085 1087 1084 1085 1087 1091 1090 1093 1093 1094 1094 1097 1093 1099 1100 1101 1100 1103 1100 1103 1106 1100 1108 1109 1110 1109 1109 1110 1114 1111 1113 1109 1095 1119 1114 1105 1089 1123 1124 1125 1126 1125 1128 1129 1128 1129 1123 1133 1134 1124 1131 1137 1137 1139 1139 1141 1141 1143 1144 1145 1146 1139 1148 1141 1150 1150 1150 1153 1151 1155 1155 1149 1158 1158 1160 1161 1162 1162 1164 1164 1160 1151 1160 1144 1133 1147 1172 1139 1174 1174 1176 1175 1174 1179 1178 1181 1182 1176 1184 1177 1175 1187 1188 1189 1189 1191 1176 1129 1194 1195 1196 1197 1198 1198 1200 1201 1201 1166 1204 1083 1206 1207 1206 1208 1210 1206 1212 1213 1195 1215 1123 1217 1217 1219 1219 1218 1222 1113 1224 1128 1226 1088 1228 1229 1230 1231 1231 1230 1228 1235 1236 1237 1238 1239 1240 1241 1241 1243 1243 1241 1243 1247 1248 1248 1248 1251 1228 1238 1230 1242 1235 1257 1258 1257 1260 1259 1262 1263 1264 1263 1263 1267 1263 1269 1270 1258 1272 1273 1274 1275 1275 1261 1278 1278 1280 1281 1282 1280 1284 1284 1281 1286 1278 1289 1290 1291 1292 1290 1294 1295 1296 1297 1298 1298 1298 1295 1295 1303 1280 1278 1274 1307 1301 1309 1279 1085 1211 1313 1313 1315 1316 1314 1318 1319 1320 1321 1322 1322 1324 1325 1326 1327 1326 1327 1330 1321 1321 1331 1334 1335 1325 1337 1338 1338 1338 1341 1341 1342 1344 1337 1342 1344 1317 1334 1339 1351 1343 1353 1211 1112 1356 1357 1358 1358 1360 1361 1361 1362 1364 1355 1366 1367 1366 1369 1369 1371 1371 1373 1374 1375 1375 1374 1378 1379 1371 1381 1381 1383 1382 1385 1386 1387 1388 1168 1390 1391 1208 1393 1394 1394 1396 1264 1398 1399 1398 1255 1327 1403 1404 1404 1405 1407 1407 1405 1409 1214 1412 1413 1412 1414 1413 1157 1418 1298 1420 1262 1422 1422 1422 1425 1425 1427 1425 1429 1155 1431 1134 1433 1434 1435 1434 1437 1369 1271 1424 1441 1220 1443 1444 1445 1444 1217 1250 1414 1450 1450 1452 1145 1454 1455 1456 1457 1458 1459 1460 1455 1457 1463 1457 1312 1236 1467 1468 1469 1470 1467 1472 1473 1472 1475 1476 1477 1477 1478 1480 1481 1481 1474 1090 1153 1289 1110 380 1489 1490 1491 1492 1491 1489 1494 1496 1497 149 1052 1368 1501 1501 1503 1503 1501 1505 1503 1508 1504 1510 1511 1512 1512 1514 1513 1512 1517 1513 1504 1507 1521 210 1523 1123 195 1526 1527 1527 1527 1530 1528 1532 1533 1534 1532 1532 1535 1538 1539 1540 1540 1542 1543 1536 1537 1544 1547 1548 1533 1550 1541 1552 1552 1552 1552 1553 1557 1558 1557 1560 1557 1554 1542 1564 1565 1555 1567 1547 1543 1531 1571 1572 1573 1572 1574 1576 1575 1571 1560 1562 1581 461 837 1584 136 1586 1586 1588 1021 1590 1591 1591 661 1594 1595 1595 1597 1598 1598 1597 1601 1602 1602 1604 1605 1602 1605 1608 1609 1596 1611 1612 1612 1613 1615 1611 1602 1618 1618 1620 1621 1621 1623 1623 1625 1626 1625 1628 1628 1624 1620 1632 1633 1634 1633 1636 1616 1599 1639 1639 1639 1642 1642 1643 1645 1645 1641 1648 1648 1650 1651 1648 1641 1645 1655 1649 1657 1653 1659 1657 1661 1661 1663 1644 1665 1666 1667 1666 1669 1670 1669 1670 1673 1673 1674 1668 1652 1641 1651 1642 1645 1644 1683 1684 1685 1686 1685 1687 1689 1683 1691 1691 1692 1691 1692 1696 1697 1683 1683 1695 1603 1702 1702 1663 1079 927 629 483 1709 1709 1711 1711 1713 1714 1715 1715 1717 1713 1715 1720 1721 1722 1723 1724 1724 1724 1727 1728 1727 1729 1727 1729 1733 1728 1735 1727 1737 1738 1739 1738 1741 1742 1715 1744 1745 1736 1358 1748 1748 1356 1751 1751 1751 850 1755 1756 1757 1755 1756 1760 1535 1762 1763 1764 1765 1764 1766 1768 1769 1762 1771 1772 1771 1772 1775 1776 1771 1778 1779 1780 1771 1776 1780 1762 883 1786 1290 1788 1150 1227 1791 1792 1793 1794 1791 508 1797 1797 1799 1800 1800 1801 598 1804 1805 1806 1807 1808 1809 1808 1811 1772 1813 1814 1814 1814 1817 1817 1819 1818 1815 950 1823 1823 663 732 1827 1828 1827 1830 1831 1832 1833 1834 1835 1836 1835 1838 1839 1839 1834 1842 1843 1843 1845 1846 1847 1848 1849 1850 1851 1848 1849 1850 1855 1845 1849 1858 1858 1858 1861 1862 1863 1864 1865 1865 1863 1863 1869 1870 1871 1861 1873 1867 1874 1876 1876 1876 1877 1877 1881 1881 1876 1881 1883 1883 1886 1888 1879 1890 1891 1892 1866 1877 1866 1896 1864 1838 1899 1899 1901 1897 1903 1903 1905 1886 1830 1908 1909 1910 1873 1883 1913 1913 1915 1913 1917 1918 1917 1919 1919 1918 1923 1922 1917 1926 1927 1927 1929 1930 1909 1904 1854 1934 1935 1934 1937 1938 1938 1940 1941 1940 1943 1937 1945 1945 1945 1945 1938 1944 1948 1935 1953 865 1955 1956 1957 1955 1959 1960 230 1962 1938 1032 1624 1966 1966 1968 1110 1970 1971 1971 1015 1729 707 711 1977 105 1979 1980 1979 1980 1981 1984 1982 711 1987 1988 1989 1990 1990 1992 1991 1994 1992 1996 1997 1997
999057 998951 998753 998558 998491 998150 997107 996934 996392 995393 994740 994043 992723 993693 992867 992772 992042 991632 991414 992613 990014 990432 990354 989903 990011 987431 984466 986876 986651 989291 984264 983813 981629 981550 974971 983535 981364 976341 963757 963140 963011 973388 973015 962440 962134 959654 968792 971546 969883 976011 975230 958170 954760 975556 951504 947726 944454 934819 938824 935705 923176 932946 932074 947181 954543 981066 933596 944085 931995 943473 942788 939567 938442 941538 925893 940716 937943 934501 925695 912072 903623 902099 892308 884678 876566 864106 892692 872201 870823 865507 845386 870807 870042 925419 923770 915052 901130 898281 899810 889642 885556 884665 869979 908570 887799 865339 827269 893650 891405 877090 893921 857809 912801 906100 907255 905149 900482 912123 881891 871590 858022 857800 848373 842356 841480 828847 837040 830778 826245 829143 817973 809447 839335 816741 834229 827585 808785 791968 788963 816231 813196 809433 777769 770048 773297 741600 688124 734564 720733 812066 805737 794186 803303 760085 795967 795568 787098 772728 752001 802120 792903 784882 754825 762592 715313 700906 701763 738364 725337 708318 698514 679625 711508 812841 679135 773287 764024 693322 771285 826529 820996 819453 807366 809565 809339 798929 817707 742424 706695 644131 639724 628103 676415 654808 610158 571557 559032 584943 564418 753424 752782 554073 740754 748691 728345 694295 665902 726948 711646 615834 602514 736481 727046 733769 713863 579952 708499 707735 690847 564935 706224 846030 800255 788618 613070 883146 886850 880245 877749 846628 877925 879897 872395 911649 856711 802001 809088 796623 808570 791398 790142 789989 787888 771260 732186 682137 766658 765888 802588 772128 802462 770498 540089 487760 515096 829458 755653 729128 778619 772096 766267 750791 745603 745347 726156 720812 706823 760364 742270 720759 707695 668965 760596 750909 729062 748633 726017 729252 706583 692272 728483 683229 681371 670928 687223 681613 656808 634643 725716 729766 675103 910295 905861 874441 852050 845567 840406 850030 843386 782780 848053 847704 811840 838948 819542 812046 813354 839229 828503 801801 792281 792172 781574 778694 828321 797071 760655 720440 736592 709395 692704 748180 673368 674080 615950 612187 610217 583789 494422 611166 609782 604026 609577 590159 585748 607422 596830 533642 612280 567466 546016 495798 609000 597356 557260 486438 525099 491551 515281 325464 505539 511361 239923 203255 149827 149006 141454 54559 72522 58537 55422 443425 396037 381873 463770 402783 589273 528894 575277 500595 491358 491263 465010 428711 368179 307855 281834 209236 286987 262836 232309 275541 256809 199849 121025 200258 250382 159160 243741 239910 490588 461175 383694 412738 343369 430511 345135 456427 414770 412853 302462 247305 584363 309575 548872 18711 489333 482134 373618 371211 336603 370641 298990 87401 607 53203 53586 457267 735088 795706 785796 783238 785658 776186 748726 767997 625238 501476 493772 460845 444729 432211 460440 453997 395086 426463 376048 360611 241929 207388 393192 375452 484577 466489 389253 429406 401436 395440 427462 419598 402703 355617 342901 315647 320036 310698 453967 550296 487707 465716 701017 696771 673496 637295 696474 586902 690078 732970 725118 612322 570620 305245 766739 763858 745369 714675 760966 694645 759543 737911 750376 730858 729171 720743 694538 707682 700923 698075 640329 700727 681493 695024 690397 731367 702377 759484 766715 759793 747987 525845 508192 488053 451089 421206 446983 415057 404120 425106 414343 301810 233165 278520 397480 340574 404402 279915 257952 228336 219533 389980 377585 302129 191461 107934 71079 244828 100832 81554 71029 208177 233417 220006 22615 388603 967261 965915 963301 963227 965588 964939 965137 370537 359673 365536 635796 813429 807785 807668 807269 788541 768336 682609 673465 763902 622582 792104 799371 789429 791660 789365 771503 750092 737222 710654 695641 717544 766228 752594 722962 714695 668860 645179 659056 788948 712950 654134 646243 624359 615325 522261 596899 561350 527504 543431 551232 516089 576607 567923 499023 497367 481652 454211 546182 450766 442139 485203 417744 407688 380820 288731 204356 159089 156137 151276 323458 144550 3439 697535 678523 667505 666344 193814 698473 788569 761739 649609 618313 611408 590346 564603 548793 912092 911418 909250 903886 903988 885149 875856 872989 854623 884385 865161 853570 824281 874385 832175 826617 826428 817482 810045 797782 826146 719196 688639 668418 687600 674113 685805 657460 674976 664262 656841 648375 627781 611750 636416 632631 636568 623688 621071 672736 620140 558676 460283 440566 434992 407038 457985 642017 640427 612560 566297 561995 559816 553649 635100 449427 607771 532084 36055 853327 859073 846723 368495 824931 792730 786572 610536 632782 501196 246254 186997 836449 402143 388105 365321 358959 297252 704779 680298 900590 892197 897433 890429 705861 745207 738828 675016 673819 657289 668887 652307 638129 636032 606992 688647 667892 753057 183847 14761 716830 903146 329140 326880 321356 259634 235835 39665 954562 948516 980135 979533 974109 977417 979425 978788 810868 665125 384434 231992 353259 197017 197770 318504 318028 288517 210690 209737 100789 186757 214669 199261 160631 116733 180060 139259 144984 885664 665010 737998 733732 727765 872921 872459 868622 860287 860861 759441 753528 731878 697250 654772 635963 599467 596734 635608 623984 614600 632476 619301 615191 584185 464014 634654 633981 631312 593402 585714 572384 564705 574572 544464 519173 496494 494625 445272 417751 507297 503458 555105 422861 437634 383316 488124 430072 413167 423194 647637 638140 629948 619557 603517 589597 569845 550363 599510 788042 375677 183762 60625 359511 11489 671792 663001 656994 637636 654757 653615 595252 636086 615268 786627 785899 765011 684667 760289 684436 648649 683791 747394 605721 532565 749406 483671 480637 442119 788552 780573 806720 802267 797513 803075 713810 698995 664985 655980 648763 635546 592341 554908 654363 645720 191599 73141 22917 328255 320785 231778 196471 197004 177670 186466 207065 203352 108705 51243 99340 140500 131907 77188 85453 60440 32987 143614 65269 60224 81467 71547 45673 124876 78691 23375 696027 651552 646855 634525 612021 613182 610296 632991 623643 616223 607269 569196 443584 587981 415154 398469 590143 607906 597499 606359 600640 562728 573571 569924 550343 455191 521124 546593 414365 544404 536305 532815 432380 543724 530837 517072 320070 517011 457105 442857 436231 514154 377841 436784 340109 401499 352411 439894 414902 427689 573432 418595 505783 479624 463696 463792 302657 457103 406812 432692 384694 298389 340323 361289 320453 241830 308794 270945 229810 164120 80335 153300 117387 119175 115200 88834 70296 124891 78284 22325 32923 1285 244761 121691 66129 25945 97965 71860 24887 13079 352367 223604 141979 881405 715819 693594 210419 275552 817160 227129 177650 124725 115816 48185 122664 172033 819903 961912 580015 575098 559958 555754 555985 534385 436727 413077 400200 359067 374175 540754 487739 476226 460866 438820 460748 383174 486558 475733 437525 426772 393567 392769 219255 174447 409901 324144 345528 218306 171329 89846 85403 26550 9011 329518 322166 298383 255851 248399 146151 135193 125633 122097 81585 451914 441430 404633 398579 323741 389352 306921 282262 223814 248645 244410 219444 189906 98433 396930 459960 311835 319223 780224 740647 720455 665498 599321 649498 723658 695270 657273 681641 648464 639466 622523 613306 612583 596607 586388 681501 679032 634235 662956 660502 650119 639022 641433 637792 656138 647734 627729 601465 639641 614930 635295 610085 599885 576637 638851 633178 622485 615102 568558 555653 621933 564221 538859 588027 569038 517191 489776 471019 522590 511810 442440 453857 519927 539359 621386 612472 499429 606521 636023 587898 525635 517555 488608 488306 486395 475132 457506 459232 566416 548797 478950 583518 401016 136439 397153 318091 386959 379255 331895 263587 263381 110425 17288 390992 376743 371813 329909 305783 363126 220150 172264 123898 139065 286731 268598 250404 246884 176283 142611 54150 44580 40114 240397 326372 218044 259552 477463 3176 994 286640 136468 259983 259689 131193 282368 240208 118236 87661 43728 259877 239950 110929 107611 56089 21521 9708 16299 5196 204578 482627 481575 476146 459155 448036 411239 439290 335130 327808 176594 24086 13221 648796 626815 581591 631885 576939 565919 621980 600594 595221 439457 403666 575565 555584 481344 422856 460927 528921 522727 492861 321089 392233 359753 567726 512640 504780 379601 349037 242112 467050 550409 545939 459646 422562 392376 357758 350779 342752 343093 78032 324077 321792 324462 276249 161125 156533 236040 159740 562160 379953 492393 337563 549358 546932 524908 506622 416546 443345 366436 354146 337973 259952 348276 295025 334146 325349 299254 534635 506351 484700 459189 458405 412082 414861 381025 386690 352311 351752 208715 250178 218085 247978 207817 120282 353562 237308 171097 160727 119560 141673 136833 127655 95830 77266 30081 33024 27122 105543 108301 57190 345867 393284 482419 476340 22135 20584 369672 638817 561996 557907 559895 557426 496469 556771 545001 501854 429886 392198 380722 372132 364692 288355 279439 264503 285610 237999 222266 385012 395730 216908 182644 135189 308502 248431 215367 115252 229762 227650 185341 125827 86985 259577 165258 73867 487738 196357 56913 19441 181242 101631 560755 419058 385738 352061 147409 314161 229332 194441 217735 167314 132410 541217 508335 505005 536137 524335 466833 408643 350332 342236 296444 296136 194616 303501 228807 219339 460945 436517 377639 300962 342790 321559 297644 290999 207859 276541 269542 185319 577390 566827 558105 553231 517425 298350 285389 280416 278010 389704 163058 139142 128545 102020 98533 42114 84077 79805 41575 591386 570281 539101 545707 527932 542931 83558 81874 4045 3794 439826 351028 427384 408464 294432 322567 297661 404788 216093 153317 111397 543074 538439 436773 365906 535764 524278 475818 288807 409269 367325 378477 373112 292098 285817 227838 567154 147272 504513 433022 499865 462227 246655 211064 119169 113745 112400 72612 32581 30360 52736 107038 105635 84220 528814 516765 504843 469822 402630 296052 361936 359148 311297 218869 164180 163579 95655 154103 92899 77127 3220 75287 145440 643916 302325 329696 515177 207878 165273 141036 93227 6517 109606 154855 71426 37991 2312 811712 276342 483331 211194 482875 446101 392144 481077 388264 453616 386331 432379 411033 317427 247758 273766 269603 235909 187701 17352 184844 423865 279342 259202 564513 523527 538567 546221 543357 529810 536505 514972 478366 525127 495865 481774 468387 502885 524312 406136 387817 311319 302238 305462 229998 196792 491950 499015 128467 50280 38773 493858 476355 297783 229963 271507 278670 240609 218709 132082 114397 143908 93394 217752 262470 276144 246836 208859 274193 258814 126015 197406 436371 436080 347946 334547 391417 318284 287413 360317 233340 4202 197405 193345 666325 643957 613045 784602 782588 759835 752349 396119 314797 280889 304446 470996 465295 457806 459010 435838 415025 427060 429001 420397 126181 229088 131153 117563 405950 10984 7821 4303 447057 419703 392775 324728 383801 365455 443395 228158 148979 192773 151398 40084 139269 104738 136810 77270 72267 113728 38887 41658 6711 156105 129148 117697 8158 107319 68236 353508 355466 335268 327679 322882 230244 286844 201126 57943 196972 285255 187886 210403 206558 69298 237600 298762 163682 11225 187038 121156 79328 30294 175697 157903 133762 75825 229086 207769 167688 160018 206825 197053 156704 78630 117327 69774 12810 32862 34955 38254 261977 21939 302243 16138 244965 206979 195341 147731 122730 150665 122000 20065 182599 96135 130695 36500 148359 92159 81560 45392 175389 145016 67841 107230 80915 66857 36110 499817 510828 852307 700088 687921 688616 668929 684886 641188 613763 610344 599194 598396 653022 556748 548177 509851 495208 333220 323968 304819 280565 279196 221843 266892 103396 268803 160298 13120 243743 171283 240353 220639 186931 179583 208937 204176 201829 552667 533599 507722 76414 243168 239001 176035 385044 283390 378367 366668 791654 763939 761870 747537 761135 761973 739239 253997 182517 178603 167300 147088 156962 146051 144689 130667 221379 203606 98005 197135 176558 115200 82719 217433 118441 60399 9390 172656 71476 57875 68625 136060 99 228216 179874 363653 340425 329881 323468 156987 13836 333503 364402 363982 95865 54966 52205 26133 2220 202457 178041 155840 111716 110336 54656 10499 101661 46492 202539 143659 90290 53943 112258 41214 82548 80766 3088 46641 283696 235824 253244 455821 970998 969226 968070 968280 965186 964366 963820 962417 958966 956516 955058 957304 945200 941447 933082 957623 953023 952646 946319 940158 939349 935324 933187 931367 927847 927650 927768 932976 931329 928454 935202 929735 919884 926998 926271 925015 906602 905457 905294 881012 901413 906270 905749 896389 894572 889687 911151 903454 873911 893006 877492 884248 890012 838982 849629 844882 845536 884432 837554 845297 838424 837648 829061 882747 874779 871746 866523 874819 877365 864529 859111 905220 955203 953802 944332 943670 849360 840527 843598 761235 834868 968271 967941 967729 966562 911128 842293 818046 832753 815898 842161 809949 789913 831830 784283 786791 809610 808186 739239 831491 820191 819761 806096 803532 802682 967904 834249 932356 929701 925725 929563 923860 920775 919542 919423 903805 909380 905938 924618 915529 917853 919382 915472 917949 888568 910951 928773 926028 280243 267190 248154 225112 265732 168545 29097 869237 868382 918833 64494 57069 47940 55405 13086 490107 463304 442895 424815 424952 149858 594159 634604 598454 833623 806146 778416 827693 763664 777198 777197 827496 640379 615470 594597 593330 539457 589002 575508 538051 507557 536489 501643 455015 491689
10
56740 981359
96153 348459
777214 867607
51379 149084
2690 26804
167780 535283
239366 477750
95673 403914
650466 650870
352183 786567
