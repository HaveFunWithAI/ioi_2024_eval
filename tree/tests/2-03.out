42214045518871
72831432357696
590641773997148
38954091559748
2020663055796
127157852441461
181696136766832
72411040396563
494394810335232
267249207833336
